src/QsCompiler/Core/SymbolTable.fs

```diff
@@ -13,23 +13,102 @@ open Microsoft.Quantum.QsCompiler.Diagnostics
 open Microsoft.Quantum.QsCompiler.SymbolResolution
 open Microsoft.Quantum.QsCompiler.SyntaxExtensions
 open Microsoft.Quantum.QsCompiler.SyntaxGenerator
-open Microsoft.Quantum.QsCompiler.SyntaxTokens 
+open Microsoft.Quantum.QsCompiler.SyntaxTokens
 open Microsoft.Quantum.QsCompiler.SyntaxTree
 open Newtonsoft.Json
 
 
 /// Represents the outcome of resolving a symbol in the symbol table.
 type ResolutionResult<'T> =
-    /// The symbol resolved successfully.
+    /// A result indicating that the symbol resolved successfully.
     | Found of 'T
-    /// An unqualified symbol is ambiguous, and it is possible to resolve it to more than one namespace. Includes the
-    /// list of possible namespaces.
+    /// A result indicating that an unqualified symbol is ambiguous, and it is possible to resolve it to more than one
+    /// namespace. Includes the list of possible namespaces.
     | Ambiguous of NonNullable<string> seq
-    /// The symbol resolved to a declaration which is not accessible from the location referencing it.
+    /// A result indicating that the symbol resolved to a declaration which is not accessible from the location
+    /// referencing it.
     | Inaccessible
-    /// No declaration with that name was found.
+    /// A result indicating that no declaration with that name was found.
     | NotFound
-    
+
+
+/// Tools for processing resolution results.
+module private ResolutionResult =
```

Comment:
It looks like almost all of these are really members to ResultionResult<'T>, and one or two would be static members. What was the reason to split them into a separate module, opposed to having them be members?
Also, a better place to put these would probably be SymbolResolutions.fs before the SymbolResolution module.

Reply:
I based it loosely off of the F# Seq module, which is organized similarly. I don't think the `seq` type has any members (static or instance), either. In this situation I prefer the function style because it's easier to use with the `|>` operator.

Reply:
Ok, sounds good. I would still make sense to move it to SymbolResolution.fs.